### PR TITLE
Drop `http.DefaultClient`

### PR DESCRIPTION
**Pull request by howardjohn**
Using a single shared client risks other parts of the code (or dependencies) modifying the client and breaking things. See https://github.com/kubernetes-sigs/gateway-api/pull/1617 for an exmaple where this happened.

Also fix up paths to avoid `//debug` requests which always get a 301.

**Please provide a description of this PR:**

### CODE REVIEW

File: pkg/kube/client.go [hunk 1]
```diff
@@ -439,6 +442,9 @@ func newClientInternal(clientFactory *clientFactory, revision string) (*client,
 
 	c.portManager = defaultAvailablePort
 
+	c.http = &http.Client{
+		Timeout: time.Second * 15,
```
Comment (zhlsunshine) [inline pkg/kube/client.go:446]: One question: why should set the timeout for client here? Is there any special?

**Review by hzxuzhonghu — APPROVED** [review]

**Review by zhlsunshine — APPROVED** [review]
LGTM!

**Review by jacob-delgado — APPROVED** [review]

**Pull request merged**